packages/datadog-plugin-aws-sdk/test/kinesis.spec.js

```diff
@@ -35,117 +43,210 @@ describe('Kinesis', () => {
+          expect(span.meta).to.have.property('streamname', streamName)
+        }).then(done, done)
+
+        helpers.putTestRecord(kinesis, streamName, helpers.dataBuffer, e => e && done(e))
+      })
+
+      describe('Disabled', () => {
+        before(() => {
+          process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED = 'false'
+        })
+
+        after(() => {
+          delete process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED
         })
-        expect(span.resource).to.equal('putRecord MyStream')
-        expect(span.meta).to.have.property('streamname', 'MyStream')
-      }).then(done, done)
 
-      helpers.putTestRecord(kinesis, helpers.dataBuffer, e => e && done(e))
+        it('skip injects trace context to Kinesis putRecord when disabled', done => {
+          helpers.putTestRecord(kinesis, streamName, helpers.dataBuffer, (err, data) => {
+            if (err) return done(err)
+
+            helpers.getTestData(kinesis, streamName, data, (err, data) => {
+              if (err) return done(err)
+
+              expect(data).not.to.have.property('_datadog')
+
+              done()
+            })
+          })
+        })
+      })
     })
 
-    describe('Disabled', () => {
+    describe('DSM Context Propagation', () => {
       before(() => {
-        process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED = 'false'
+        return agent.load('aws-sdk', { kinesis: { dsmEnabled: true } }, { dsmEnabled: true })
       })
 
-      after(() => {
-        delete process.env.DD_TRACE_AWS_SDK_KINESIS_ENABLED
+      before(done => {
+        tracer = require('../../dd-trace')
+        tracer.use('aws-sdk', { kinesis: { dsmEnabled: true } }, { dsmEnabled: true })
+
+        createResources(streamNameDSM, done)
+      })
+
+      after(done => {
+        kinesis.deleteStream({
+          StreamName: streamNameDSM
+        }, (err, res) => {
+          if (err) return done(err)
+
+          helpers.waitForDeletedStream(kinesis, streamNameDSM, done)
+        })
       })
 
-      it('skip injects trace context to Kinesis putRecord when disabled', done => {
-        helpers.putTestRecord(kinesis, helpers.dataBuffer, (err, data) => {
+      it('injects DSM pathway hash during Kinesis putRecord to the span', done => {
+        helpers.putTestRecord(kinesis, streamNameDSM, helpers.dataBuffer, (err, data) => {
           if (err) return done(err)
 
-          helpers.getTestData(kinesis, data, (err, data) => {
+          let putRecordSpanMeta = {}
+          agent.use(traces => {
+            const span = traces[0][0]
+
+            if (span.resource.startsWith('putRecord')) {
+              putRecordSpanMeta = span.meta
+            }
+
+            expect(putRecordSpanMeta).to.include({
+              'pathway.hash': expectedProducerHash.readBigUInt64BE(0).toString()
+            })
+          }).then(done, done)
+        })
+      })
+
+      describe('emits a new DSM Stats to the agent when DSM is enabled', () => {
```

Comment:
Can this be restructured a bit? A `describe` should describe a context, and `it` should describe an expectation. The current structure is counter-intuitive.

Reply:
Can you take another look, the context in this case would be `DSM Stats`, with the expectation being that they should be created for consume/produce operations

Reply:
Not sure I understand, but also, that's not how these are structured. It should be something like `describe('when some statement')`, describing the context and the setup, and `it('should do something')` to describe the expected behaviour of the test itself. Another structure doesn't properly describe what the test is doing.

Reply:
TBH the only reason I added this nested describe block was to share the `before` setup with both tests. Where is the before setup we perform a produce and consume, then assert that the stats were created in the tests. Not sure how I should properly structure these tests otherwise.

Reply:
I have changed the test structure to be `describe(when dsm is enabled)` followed by all tests.